### PR TITLE
Bazel fixes

### LINKED ISSUE #9520: Threading compiler errors
### What is the problem?

```
ray/rpc/worker/core_worker_client.h:332:19: error: 'guarded_by' attribute requires arguments whose type is annotated with 'capability' attribute; type here is 'std::mutex'
      send_queue_ GUARDED_BY(mutex_);
ray/rpc/worker/core_worker_client.h:335:32: error: 'guarded_by' attribute requires arguments whose type is annotated with 'capability' attribute; type here is 'std::mutex'
  int64_t rpc_bytes_in_flight_ GUARDED_BY(mutex_) = 0;
ray/rpc/worker/core_worker_client.h:338:32: error: 'guarded_by' attribute requires arguments whose type is annotated with 'capability' attribute; type here is 'std::mutex'
  int64_t max_finished_seq_no_ GUARDED_BY(mutex_) = -1;
ray/rpc/worker/core_worker_client.h:262:7: error: reading variable 'send_queue_' requires holding mutex 'mutex_'
      send_queue_.push_back(std::make_pair(std::move(request), callback));
ray/rpc/worker/core_worker_client.h:299:24: error: reading variable 'max_finished_seq_no_' requires holding mutex 'mutex_'
          if (seq_no > max_finished_seq_no_) {
ray/rpc/worker/core_worker_client.h:300:13: error: writing variable 'max_finished_seq_no_' requires holding mutex 'mutex_' exclusively
            max_finished_seq_no_ = seq_no;
ray/rpc/worker/core_worker_client.h:302:11: error: writing variable 'rpc_bytes_in_flight_' requires holding mutex 'mutex_' exclusively
          rpc_bytes_in_flight_ -= task_size;
ray/rpc/worker/core_worker_client.h:303:21: error: reading variable 'rpc_bytes_in_flight_' requires holding mutex 'mutex_'
          RAY_CHECK(rpc_bytes_in_flight_ >= 0);
ray/rpc/worker/core_worker_client.h:284:13: error: reading variable 'send_queue_' requires holding mutex 'mutex_'
    while (!send_queue_.empty() && rpc_bytes_in_flight_ < kMaxBytesInFlight) {
ray/rpc/worker/core_worker_client.h:284:36: error: reading variable 'rpc_bytes_in_flight_' requires holding mutex 'mutex_'
    while (!send_queue_.empty() && rpc_bytes_in_flight_ < kMaxBytesInFlight) {
ray/rpc/worker/core_worker_client.h:285:30: error: reading variable 'send_queue_' requires holding mutex 'mutex_'
      auto pair = std::move(*send_queue_.begin());
ray/rpc/worker/core_worker_client.h:286:7: error: reading variable 'send_queue_' requires holding mutex 'mutex_'
      send_queue_.pop_front();
ray/rpc/worker/core_worker_client.h:292:43: error: reading variable 'max_finished_seq_no_' requires holding mutex 'mutex_'
      request->set_client_processed_up_to(max_finished_seq_no_);
ray/rpc/worker/core_worker_client.h:293:7: error: writing variable 'rpc_bytes_in_flight_' requires holding mutex 'mutex_' exclusively
      rpc_bytes_in_flight_ += task_size;
ray/rpc/worker/core_worker_client.h:312:10: error: reading variable 'send_queue_' requires holding mutex 'mutex_'
    if (!send_queue_.empty()) {
ray/rpc/worker/core_worker_client.h:313:54: error: reading variable 'send_queue_' requires holding mutex 'mutex_'
      RAY_LOG(DEBUG) << "client send queue size " << send_queue_.size();
```

### Reproduction (REQUIRED)

Uncomment the warning suppression in #9519.

If we cannot run your script, we cannot fix your issue.

- [ ] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/latest/installation.html).

### PR DESCRIPTION
**Pull request by mehrdadn**
## Why are these changes needed?

Gets rid of most `includes = ...` in Bazel rules, which should be avoided and mask bugs as a result of transitive inclusions.

Fixes various warnings and errors that occur as a consequence.

## Related issue number

Closes #9520

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: src/ray/rpc/gcs_server/gcs_rpc_client.h [hunk 1]
```diff
@@ -17,8 +17,8 @@
 #include <unistd.h>
 
 #include "ray/common/network_util.h"
-#include "ray/protobuf/gcs_service.grpc.pb.h"
 #include "ray/rpc/grpc_client.h"
+#include "src/ray/protobuf/gcs_service.grpc.pb.h"
```
Comment (suquark) [inline src/ray/rpc/gcs_server/gcs_rpc_client.h:21]: why we have "src" prefix here

File: src/ray/raylet_client/raylet_client.h [hunk 2]
```diff
@@ -25,6 +23,7 @@
 #include "ray/common/status.h"
 #include "ray/common/task/task_spec.h"
 #include "ray/rpc/node_manager/node_manager_client.h"
+#include "src/ray/protobuf/gcs.pb.h"
```
Comment (suquark) [inline src/ray/raylet_client/raylet_client.h:26]: why we have "src" prefix here (also in other places)

File: BUILD.bazel [hunk 3]
```diff
@@ -1446,8 +1446,8 @@ cc_library(
     hdrs = [
         "src/ray/thirdparty/sha256.h",
     ],
-    copts = COPTS,
-    includes = ["src/ray/thirdparty"],
+    #copts = COPTS,
```
Comment (suquark) [inline BUILD.bazel:1449]: should we keep the comment mark (`#`)?

File: src/ray/rpc/gcs_server/gcs_rpc_client.h (hunk 1 above)
Comment (mehrdadn) [reply src/ray/rpc/gcs_server/gcs_rpc_client.h:21]: Unfortunately for protobuf headers I couldn't figure out a way to make it work without `src`.

File: src/ray/raylet_client/raylet_client.h (hunk 2 above)
Comment (mehrdadn) [reply src/ray/raylet_client/raylet_client.h:26]: See above.

File: BUILD.bazel (hunk 3 above)
Comment (mehrdadn) [reply BUILD.bazel:1449]: Oops, good catch, thanks!

**Comment by suquark** [comment]
The CI failures are not related. I'll merge it.

**Review by suquark — APPROVED** [review]

**Pull request merged**